lib/tapioca/rbi/rewriters/sort_nodes.rb

```diff
@@ -70,6 +70,15 @@ def node_name(node)
             node.name
           end
         end
+
+        sig { params(nodes: T::Array[Node]).returns(T::Hash[Node, Integer]) }
+        def original_nodes_order(nodes)
+          order = {}
+          nodes.each_with_index do |node, i|
+            order[node] = i
+          end
+          order
```

Comment:
Nitpick:
```suggestion
          nodes.each_with_index.each_with_object do |node, index, result|
            result[node] = index
          end
```

might be more idiomatic

Reply:
It would have been nice like this but:

```
irb(main):001:0> arr = ["a", "b", "c"]
=> ["a", "b", "c"]
irb(main):002:0> arr.each_with_index.each_with_object({}) { |x, i, res| res[x] = i }
(irb):2:in `block in <main>': undefined method `[]=' for nil:NilClass (NoMethodError)
	from (irb):2:in `each'
	from (irb):2:in `each_with_index'
	from (irb):2:in `each_with_object'
	from (irb):2:in `<main>'
	from /opt/rubies/3.0.1/lib/ruby/gems/3.0.0/gems/irb-1.3.5/exe/irb:11:in `<top (required)>'
	from /opt/rubies/3.0.1/bin/irb:23:in `load'
	from /opt/rubies/3.0.1/bin/irb:23:in `<main>'
```

My understanding is that we need to destructure the couple:

```rb
nodes.each_with_index.each_with_object({}) do |node_plus_index, result|
  node, index = *node_plus_index
     result[node] = index
  end
end
```

Do we still want this change?

Reply:
yeah, maybe we need `|(node, index), result|`?

Reply:
yes, that works:
```ruby
["a", "b", "c"].each_with_index.each_with_object({}) { |(x, i), res| res[x] = i }
# => {"a"=>0, "b"=>1, "c"=>2}
```

Reply:
There is also the much shorter:
```ruby
["a", "b", "c"].map.with_index.map.to_h
# => {"a"=>0, "b"=>1, "c"=>2}
```
:)

Reply:
I went with `original_order = node.nodes.map.with_index.to_h` and removed the now-useless method 👍